### PR TITLE
nix: adding nix shell support

### PR DESCRIPTION
**Pull request by wiredhikari**
shell.nix - added tools for development 
default.nix - added metadata

### CODE REVIEW

**Comment by elsirion** [comment]
Great to see progress on the nix integration!

Does the `default.nix` require any special build options/environment? I expected a plain `nix-build` to work but it doesn't (variable minimint is undefined etc., which seems correct, maybe it needs to be a string?).

Also, the owner most likely needs to be fedimint and not elsirion for the github fetch function.

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    repo = "minimint";
+    rev = "master";
+    sha256 = "sha256-uoTKfzsm9VkjZPXGoQOCPBzxMTkP8zTSf48eiv5zgAA=";
+  };
+
+  cargoSha256 =  "sha256-1lZElXK9M895DURLID2KJdmCkJRTFNVC3meLFluO2WU=";
+
+  buildInputs = [
+    # build
+   ];
```
[the remark this answers is not in the document]
Comment (wiredhikari) [reply default.nix:24]: It can be used is some areas, but we will remove it for now and add when we need it.

File: shell.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ];
+  }
+}:
+
+pkgs.mkShell {
+  packages = with pkgs; [
+    rustc
+    cargo
+    rust-analyzer
+  ];
```
Comment (elsirion) [inline shell.nix:19]: I wonder if this should include packages needed to run the integration test (at least bitcoin and clightning). Probably more useful.

**Comment by elsirion** [comment]
There seems to be a [nix GH action](https://nix.dev/tutorials/continuous-integration-github-actions), would be cool to run integration tests in the nix shell instead of installing bitcoind and c-lightning manually. As a bonus we'd also test the shell. Also testing the derivation would be nice, what do you think would be the best way?

**Comment by wiredhikari** [comment]
Sounds great to me, will setup a gh action for nix-shell and will have to figure our how to go about installing bitcoind and c-lightning, As of now,I can think of is using cachix or some docker image.

**Comment by elsirion** [comment]
Nit: please rather rebase on master than merging it into your branch like you did in 86bef22.

**Pull request closed without merging**